webpack/ForemanWebhooks/Routes/Webhooks/WebhooksIndexPage/Components/WebhookCreateModal.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+import { submitForm } from 'foremanReact/redux/actions/common/forms';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import ForemanModal from 'foremanReact/components/ForemanModal';
+
+import ConnectedWebhookForm from '../../Components/WebhookForm';
+
+import { WEBHOOK_CREATE_MODAL_ID, WEBHOOKS_PATH } from '../../constants';
+
+const WebhookCreateModal = ({ onSuccess, onCancel }) => {
+  const dispatch = useDispatch();
+
+  const initialWebhookValues = {
+    name: '',
+    target_url: '',
+    user: '',
+    password: '',
+    webhook_template_id: undefined,
+    event: '',
+    http_method: 'POST',
+    http_content_type: 'application/json',
+    enabled: true,
+    verify_ssl: true,
+    ssl_ca_certs: '',
+    http_headers: '',
+  };
+
+  const handleSubmit = (values, actions) =>
+    dispatch(
+      submitForm({
+        url: foremanUrl(`/api${WEBHOOKS_PATH}`),
+        values: { ...values, controller: 'webhooks' },
+        item: 'Webhook',
+        message: __('Webhook was successfully created.'),
+        successCallback: onSuccess,
+        actions,
+      })
+    );
+
+  return (
+    <ForemanModal id={WEBHOOK_CREATE_MODAL_ID} title={__('Create Webhook')}>
+      <ForemanModal.Header />
+      <ConnectedWebhookForm
+        handleSubmit={handleSubmit}
+        initialValues={initialWebhookValues}
+        onCancel={onCancel}
+      />
+    </ForemanModal>
+  );
+};
+
+WebhookCreateModal.propTypes = {
+  onSuccess: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
+};
+
+export default WebhookCreateModal;
```

Comment:
small issue with the modal, when accidentally clicking outside of it, all the data that the user entered is gone :(
is it possible to disable modal close after click outside? I think that the PF4 modal disables it.

![Webhooks](https://user-images.githubusercontent.com/26363699/111625808-f4479000-87f5-11eb-9aa4-eb27deb049a6.gif)